src/main/java/org/jenkinsci/plugins/casc/ConfigurationAsCode.java

```diff
@@ -238,22 +238,28 @@ public void doExport(StaplerRequest req, StaplerResponse res) throws Exception {
         }
 
         MappingNode root = new MappingNode(Tag.MAP, tuples, BLOCK.getStyleBoolean());
+        try (Writer writer = new OutputStreamWriter(res.getOutputStream(), StandardCharsets.UTF_8)) {
+            serializeYamlNode(root, writer);
+        } catch (IOException e) {
+            throw new YAMLException(e);
+        }
+    }
+
+    // public only for testing
+    public static void serializeYamlNode(Node root, Writer writer) throws IOException {
```

Comment:
Then use `@Restricted(NoExternalUse.class)`. It will prevent other plugins from using this API

Reply:
package private should be enough anyway

Reply:
I made it package private